From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anze <aanzdev@gmail.com>
Date: Tue, 17 Jun 2026 21:30:00 +0200
Subject: [PATCH] rocknix-splash: DUCKTALE boot logo (green pixel-art duck)

Cosmetic rebrand of the boot splash: replace the embedded ROCKNIX wordmark SVG
paths with a pixel-art duck + "DUCKTALE" wordmark, all in mallard green
(#1F4D2E / rgb(31,77,46)) on the existing black background. Only the
svg_paths[]/svg_colors[] arrays change; the framebuffer/rotation/render logic
(and the 1284x500 viewBox the renderer scales to) is untouched.

Signed-off-by: Anze <aanzdev@gmail.com>
---
 main.c | 219 +++++++++++++++++++++++++++++++++++++++++++++------------
 1 file changed, 175 insertions(+), 44 deletions(-)

diff --git a/main.c b/main.c
--- a/main.c
+++ b/main.c
@@ -22,28 +22,210 @@ int z9(void);
  * 5: "I" in logo
  * 6: "X" in logo
  */
+/*
+ * DUCKTALE pixel-art logo (green duck + wordmark) rendered as SVG rect
+ * paths in a 1284x500 viewBox. Cosmetic rebrand of the upstream ROCKNIX
+ * splash; the rest of the splash app is unchanged.
+ */
 const char *svg_paths[] = {
-    "M -0.002 140.106 L -0.002 359.628 L 27.138 359.628 L 27.138 270.09 L 47.626 270.09 L 115.612 359.628 L 150.077 359.628 L 77.701 267.695 C 112.29 262.506 133.445 239.357 133.445 204.499 C 133.445 164.719 105.902 140.106 62.269 140.106 L -0.002 140.106 Z M 27.138 161.925 L 54.944 161.925 C 86.613 161.925 105.368 178.024 105.368 204.499 C 105.368 230.975 86.613 247.073 54.944 247.073 L 27.138 247.073 L 27.138 161.925 Z",
-    "M 166.036 250 C 166.036 313.329 215.268 363.886 276.734 363.886 C 338.2 363.886 387.424 313.329 387.424 250 C 387.424 186.671 338.2 136.116 276.734 136.116 C 215.268 136.116 166.036 186.671 166.036 250 Z M 195.306 250 C 195.306 200.375 230.561 162.325 276.734 162.325 C 322.899 162.325 358.154 200.375 358.154 250 C 358.154 299.625 322.899 337.676 276.734 337.676 C 230.561 337.676 195.306 299.625 195.306 250 Z",
-    "M 530.178 363.886 C 557.45 363.886 582.461 355.504 600.557 341.801 L 586.99 319.715 C 572.619 331.023 552.928 337.144 531.905 337.144 C 478.957 337.144 443.965 299.093 443.965 250 C 443.965 200.907 478.957 162.857 531.905 162.857 C 552.928 162.857 572.619 168.977 586.99 180.285 L 600.557 158.2 C 582.461 144.497 557.45 136.116 530.178 136.116 C 463.254 136.116 414.432 185.874 414.432 250 C 414.432 314.128 463.254 363.886 530.178 363.886 Z",
-    "M 661.357 359.628 L 661.357 262.108 L 762.073 359.628 L 799.853 359.628 L 685.438 247.872 L 788.145 140.106 L 754.353 140.106 L 661.357 237.894 L 661.357 140.106 L 634.217 140.106 L 634.217 359.628 L 661.357 359.628 Z",
-    "M 821.944 140.106 L 821.944 359.628 L 849.084 359.628 L 849.084 186.006 L 960.038 359.628 L 987.178 359.628 L 987.178 140.106 L 960.038 140.106 L 960.038 313.861 L 849.084 140.106 L 821.944 140.106 Z",
-    "M 1034.814 140.106 L 1034.814 359.627 L 1061.955 359.627 L 1061.955 140.106 L 1034.814 140.106 Z",
-    "M 1116.9 359.628 L 1183.554 264.768 L 1250.208 359.628 L 1284 359.628 L 1200.45 241.219 L 1270.96 140.106 L 1238.369 140.106 L 1183.554 218.469 L 1128.74 140.106 L 1096.141 140.106 L 1166.658 241.219 L 1083.109 359.628 L 1116.9 359.628 Z"
+    "M 216 40 L 320 40 L 320 66 L 216 66 Z",
+    "M 164 66 L 372 66 L 372 92 L 164 92 Z",
+    "M 138 92 L 398 92 L 398 118 L 138 118 Z",
+    "M 138 118 L 242 118 L 242 144 L 138 144 Z",
+    "M 268 118 L 398 118 L 398 144 L 268 144 Z",
+    "M 138 144 L 398 144 L 398 170 L 138 170 Z",
+    "M 424 144 L 502 144 L 502 170 L 424 170 Z",
+    "M 138 170 L 398 170 L 398 196 L 138 196 Z",
+    "M 424 170 L 502 170 L 502 196 L 424 196 Z",
+    "M 138 196 L 398 196 L 398 222 L 138 222 Z",
+    "M 164 222 L 372 222 L 372 248 L 164 248 Z",
+    "M 164 248 L 346 248 L 346 274 L 164 274 Z",
+    "M 138 274 L 372 274 L 372 300 L 138 300 Z",
+    "M 112 300 L 398 300 L 398 326 L 112 326 Z",
+    "M 86 326 L 424 326 L 424 352 L 86 352 Z",
+    "M 60 352 L 424 352 L 424 378 L 60 378 Z",
+    "M 60 378 L 424 378 L 424 404 L 60 404 Z",
+    "M 86 404 L 424 404 L 424 430 L 86 430 Z",
+    "M 138 430 L 372 430 L 372 456 L 138 456 Z",
+    "M 560 201 L 616 201 L 616 215 L 560 215 Z",
+    "M 560 215 L 574 215 L 574 229 L 560 229 Z",
+    "M 616 215 L 630 215 L 630 229 L 616 229 Z",
+    "M 560 229 L 574 229 L 574 243 L 560 243 Z",
+    "M 616 229 L 630 229 L 630 243 L 616 243 Z",
+    "M 560 243 L 574 243 L 574 257 L 560 257 Z",
+    "M 616 243 L 630 243 L 630 257 L 616 257 Z",
+    "M 560 257 L 574 257 L 574 271 L 560 271 Z",
+    "M 616 257 L 630 257 L 630 271 L 616 271 Z",
+    "M 560 271 L 574 271 L 574 285 L 560 285 Z",
+    "M 616 271 L 630 271 L 630 285 L 616 285 Z",
+    "M 560 285 L 616 285 L 616 299 L 560 299 Z",
+    "M 650 201 L 664 201 L 664 215 L 650 215 Z",
+    "M 706 201 L 720 201 L 720 215 L 706 215 Z",
+    "M 650 215 L 664 215 L 664 229 L 650 229 Z",
+    "M 706 215 L 720 215 L 720 229 L 706 229 Z",
+    "M 650 229 L 664 229 L 664 243 L 650 243 Z",
+    "M 706 229 L 720 229 L 720 243 L 706 243 Z",
+    "M 650 243 L 664 243 L 664 257 L 650 257 Z",
+    "M 706 243 L 720 243 L 720 257 L 706 257 Z",
+    "M 650 257 L 664 257 L 664 271 L 650 271 Z",
+    "M 706 257 L 720 257 L 720 271 L 706 271 Z",
+    "M 650 271 L 664 271 L 664 285 L 650 285 Z",
+    "M 706 271 L 720 271 L 720 285 L 706 285 Z",
+    "M 664 285 L 706 285 L 706 299 L 664 299 Z",
+    "M 754 201 L 810 201 L 810 215 L 754 215 Z",
+    "M 740 215 L 754 215 L 754 229 L 740 229 Z",
+    "M 740 229 L 754 229 L 754 243 L 740 243 Z",
+    "M 740 243 L 754 243 L 754 257 L 740 257 Z",
+    "M 740 257 L 754 257 L 754 271 L 740 271 Z",
+    "M 740 271 L 754 271 L 754 285 L 740 285 Z",
+    "M 754 285 L 810 285 L 810 299 L 754 299 Z",
+    "M 830 201 L 844 201 L 844 215 L 830 215 Z",
+    "M 886 201 L 900 201 L 900 215 L 886 215 Z",
+    "M 830 215 L 844 215 L 844 229 L 830 229 Z",
+    "M 872 215 L 886 215 L 886 229 L 872 229 Z",
+    "M 830 229 L 844 229 L 844 243 L 830 243 Z",
+    "M 858 229 L 872 229 L 872 243 L 858 243 Z",
+    "M 830 243 L 858 243 L 858 257 L 830 257 Z",
+    "M 830 257 L 844 257 L 844 271 L 830 271 Z",
+    "M 858 257 L 872 257 L 872 271 L 858 271 Z",
+    "M 830 271 L 844 271 L 844 285 L 830 285 Z",
+    "M 872 271 L 886 271 L 886 285 L 872 285 Z",
+    "M 830 285 L 844 285 L 844 299 L 830 299 Z",
+    "M 886 285 L 900 285 L 900 299 L 886 299 Z",
+    "M 920 201 L 990 201 L 990 215 L 920 215 Z",
+    "M 948 215 L 962 215 L 962 229 L 948 229 Z",
+    "M 948 229 L 962 229 L 962 243 L 948 243 Z",
+    "M 948 243 L 962 243 L 962 257 L 948 257 Z",
+    "M 948 257 L 962 257 L 962 271 L 948 271 Z",
+    "M 948 271 L 962 271 L 962 285 L 948 285 Z",
+    "M 948 285 L 962 285 L 962 299 L 948 299 Z",
+    "M 1024 201 L 1066 201 L 1066 215 L 1024 215 Z",
+    "M 1010 215 L 1024 215 L 1024 229 L 1010 229 Z",
+    "M 1066 215 L 1080 215 L 1080 229 L 1066 229 Z",
+    "M 1010 229 L 1024 229 L 1024 243 L 1010 243 Z",
+    "M 1066 229 L 1080 229 L 1080 243 L 1066 243 Z",
+    "M 1010 243 L 1080 243 L 1080 257 L 1010 257 Z",
+    "M 1010 257 L 1024 257 L 1024 271 L 1010 271 Z",
+    "M 1066 257 L 1080 257 L 1080 271 L 1066 271 Z",
+    "M 1010 271 L 1024 271 L 1024 285 L 1010 285 Z",
+    "M 1066 271 L 1080 271 L 1080 285 L 1066 285 Z",
+    "M 1010 285 L 1024 285 L 1024 299 L 1010 299 Z",
+    "M 1066 285 L 1080 285 L 1080 299 L 1066 299 Z",
+    "M 1100 201 L 1114 201 L 1114 215 L 1100 215 Z",
+    "M 1100 215 L 1114 215 L 1114 229 L 1100 229 Z",
+    "M 1100 229 L 1114 229 L 1114 243 L 1100 243 Z",
+    "M 1100 243 L 1114 243 L 1114 257 L 1100 257 Z",
+    "M 1100 257 L 1114 257 L 1114 271 L 1100 271 Z",
+    "M 1100 271 L 1114 271 L 1114 285 L 1100 285 Z",
+    "M 1100 285 L 1170 285 L 1170 299 L 1100 299 Z",
+    "M 1190 201 L 1260 201 L 1260 215 L 1190 215 Z",
+    "M 1190 215 L 1204 215 L 1204 229 L 1190 229 Z",
+    "M 1190 229 L 1204 229 L 1204 243 L 1190 243 Z",
+    "M 1190 243 L 1246 243 L 1246 257 L 1190 257 Z",
+    "M 1190 257 L 1204 257 L 1204 271 L 1190 271 Z",
+    "M 1190 271 L 1204 271 L 1204 285 L 1190 285 Z",
+    "M 1190 285 L 1260 285 L 1260 299 L 1190 299 Z"
 };
 
-/* Color definitions for each path component
- * First 4 paths are red (brand color)
- * Last 3 paths are gray (secondary color)
- */
+/* Mallard green (#1F4D2E) for every path. */
 const char *svg_colors[] = {
-    "rgb(255,85,85)",  // Red
-    "rgb(255,85,85)",  // Red
-    "rgb(255,85,85)",  // Red
-    "rgb(255,85,85)",  // Red
-    "rgb(85,85,85)",   // Gray
-    "rgb(85,85,85)",   // Gray
-    "rgb(85,85,85)"    // Gray
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)",
+    "rgb(31,77,46)"
 };
 
 #define NUM_PATHS (sizeof(svg_paths) / sizeof(svg_paths[0]))
